File: src/models/song.model.js
const mongoose = require('mongoose');
const Schema = mongoose.Schema;

const songSchema = new Schema({
    title: String,
    artist: String,
    album: String,
    duration: Number,
    genre: String,
    yearReleased: Number,
    bitRate: Number,
    encoder: String,
    path: String,
    createdAt: Date,
    updatedAt: Date
});

const SongModel = mongoose.model('Song', songSchema, 'songs');
module.exports = SongModel;

File: src/main/java/server/models/SongModel.java
package server.models;

import org.bson.types.ObjectId;

import java.time.LocalDateTime;
import java.util.Date;

public class SongModel {
    public ObjectId _id;
    public String artist;
    public String album;
    public String composer;
    public String genre;
    public String title;
    public int year;
    public String bitrate;
    public String filePath;
    public Date updatedAt;
}
